src/structures/User.js

```diff
@@ -56,12 +59,14 @@ class User extends Base {
       this.username = null;
     }
 
-    if ('bot' in data || typeof this.bot !== 'boolean') {
+    if ('bot' in data || typeof this.bot === 'boolean') {
       /**
        * Whether or not the user is a bot
-       * @type {boolean}
+       * @type {?boolean}
        */
-      this.bot = Boolean(data.bot);
+      this.bot = Boolean(data.bot ?? this.bot);
+    } else if (!this.partial) {
```

Comment:
wouldn't it make more sense to
`else if (!this.partial && typeof this.bot !== 'boolean')` instead

Reply:
Would this replace the typeof this.bot check above or is it just an additional check?

Reply:
it would replace it

Reply:
Technically both work, why do you prefer this approach?

Reply:
because its essentially just adding `this.bot = Boolean(this.bot)` for no reason, just makes more sense

Reply:
There are instances where data.bot does have a value, but I understand what you're saying so I'll change that